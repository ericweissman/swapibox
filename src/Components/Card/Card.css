.people-card {
  width: 250px;
  height: 285px;
  color: yellow;
  margin: 20px;
  border: 3px solid yellow;
  border-radius: 4px;
  background-color: #101111;
}
/*# sourceMappingURL=Card.css.map */