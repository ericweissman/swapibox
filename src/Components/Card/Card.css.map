{
    "version": 3,
    "mappings": "AAAA,AAAA,YAAY,CAAC;EACX,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,MAAM;EACb,MAAM,EAAE,IAAI;EACZ,MAAM,EAAE,gBAAgB;EACxB,aAAa,EAAE,GAAG;EAClB,gBAAgB,EAAE,OAAO;CAC1B",
    "sources": [
        "Card.scss"
    ],
    "names": [],
    "file": "Card.css"
}