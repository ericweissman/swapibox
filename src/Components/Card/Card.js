import React from 'react';
import '../../Main.scss';
import PropTypes from 'prop-types'

const Card = (props) => {
  let { people, planets, vehicles } = props.category
  switch (props.active) {
    case 'people':
      return people.map((person) => {
        return (
          <div className="card people-card">
            <h3>{person.name}</h3>
            <p>Homeworld: {person.homeworld}</p>
            <p>Population: {person.population}</p>
            <p>Species: {person.species}</p>
            <button>Add to Favorites</button>
          </div>
        )
      });
      break;

    case 'planets':
      return planets.map((planet) => {
        return (
          <div className='card planet-card'>
            <h3>{planet.name}</h3>
            <p>Terrain: {planet.terrain}</p>
            <p>Population: {planet.population}</p>
            <p>Climate: {planet.climate}</p>
            <ul>
              Residents: {
                planet.residents.map((resident) => {
                  return (
                    <li>{resident}</li>
                  )
                })
              }
            </ul>
          </div>
        )
      })
    case 'vehicles':
      return vehicles.map((vehicle) => {
        return (
          <div className="card vehicle-card">
            <h3>{vehicle.name}</h3>
            <p>Model: {vehicle.model}</p>
            <p>Class: {vehicle.class}</p>
            <p>Passengers: {vehicle.passengers}</p>
            <button>Add to Favorites</button>
          </div>
        )
      })
  }
}

Card.propTypes = {
  active: PropTypes.string,
  category: PropTypes.shape({
    films: PropTypes.object,
    people: PropTypes.array,
    planets: PropTypes.array,
    vehicles: PropTypes.array,
    active: PropTypes.string,
    errorStatus: PropTypes.errorStatus
  })
}

export default Card
