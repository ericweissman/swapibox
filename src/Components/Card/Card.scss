.card {
  width: 250px;
  height: 285px;
  color: $text;
  margin: 20px;
  border: 3px solid $accent;
  border-radius: 4px;
  background-color: #101111;
  text-transform: uppercase;
}
// .people-card {
//   width: 250px;
//   height: 285px;
//   color: $text;
//   margin: 20px;
//   border: 3px solid $accent;
//   border-radius: 4px;
//   background-color: #101111;
//   text-transform: uppercase;
// }

// .planet-card {
//   width: 250px;
//   height: 285px;
//   color: $text;
//   margin: 20px;
//   border: 3px solid $accent;
//   border-radius: 4px;
//   background-color: #101111;
//   text-transform: uppercase;

// }