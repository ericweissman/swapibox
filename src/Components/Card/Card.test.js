import React from 'react';
import Card from './Card'
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  let activeMock;
  const categoryMock = {
    people: [{ name: 'a', homeworld: 'b', population: 'c', species: 'd' }],
    planets: [{ name: 'a', terrain: 'b', population: 'c', climate: 'd', residents: ['lile'] }],
    vehicles: [ { name: 'a', class: 'b', passengers: 'c'}]
  }
  
  it('should match the correct snapshot when people is passed as props', () => {
    activeMock = 'people';
    wrapper = shallow(<Card active={activeMock} category={categoryMock} />)
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the correct snapshot when planets is passed as props', () => {
    activeMock = 'planets';
    wrapper = shallow(<Card active={activeMock} category={categoryMock} />)
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the correct snapshot when vehicles is passed as props', () => {
    activeMock = 'vehicles';
    wrapper = shallow(<Card active={activeMock} category={categoryMock} />)
    expect(wrapper).toMatchSnapshot();
  })
})