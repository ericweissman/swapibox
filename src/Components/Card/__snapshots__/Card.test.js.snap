// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the correct snapshot when people is passed as props 1`] = `
<div
  className="card people-card"
>
  <h3>
    a
  </h3>
  <p>
    Homeworld: 
    b
  </p>
  <p>
    Population: 
    c
  </p>
  <p>
    Species: 
    d
  </p>
  <button>
    Add to Favorites
  </button>
</div>
`;

exports[`Card should match the correct snapshot when planets is passed as props 1`] = `
<div
  className="card planet-card"
>
  <h3>
    a
  </h3>
  <p>
    Terrain: 
    b
  </p>
  <p>
    Population: 
    c
  </p>
  <p>
    Climate: 
    d
  </p>
  <ul>
    Residents: 
    <li>
      lile
    </li>
  </ul>
</div>
`;

exports[`Card should match the correct snapshot when vehicles is passed as props 1`] = `
<div
  className="card vehicle-card"
>
  <h3>
    a
  </h3>
  <p>
    Model: 
  </p>
  <p>
    Class: 
    b
  </p>
  <p>
    Passengers: 
    c
  </p>
  <button>
    Add to Favorites
  </button>
</div>
`;
